File: leetcode/Arrays/leetcode-0066/66.go
package arrays

func plusOne(digits []int) []int {
	if digits == nil || len(digits) == 0 {
		return digits
	}
	plus := 1
	for i := len(digits) - 1; i >= 0; i-- {
		digits[i] += plus
		plus = digits[i] / 10
		digits[i] %= 10
	}
	if plus > 0 {
		digits = append([]int{plus}, digits...)
	}
	return digits
}

File: leetcode/Arrays/leetcode-0066/66.java
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits == null || digits.length == 0) {
            return digits;
        }
        int plus = 1;
        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i] = digits[i] + plus;
            plus = digits[i] / 10;
            digits[i] %= 10;
        }
        if (plus > 0) {
            int[] newDigits = new int[digits.length + 1];
            newDigits[0] = plus;
            for (int i = 0; i < digits.length; i++) {
                newDigits[i + 1] = digits[i];
            }
            digits = newDigits;
        }
        return digits;
    }
}